doc: improve PM config info (#15018)
nrfconnect/sdk-nrf: scripts/partition_manager/partition_manager.rst

.. note::
  If the static configuration contains an entry for the ``app`` partition, this entry is ignored.

.. _ug_pm_config_dynamic:

Configuring dynamic partitions
==============================

For child images that use dynamic partition sizing, the Partition Manager allows you to use the :kconfig:option:`CONFIG_PM_PARTITION_SIZE_<CHILD_IMAGE>` Kconfig option to adjust the partition size.
This is particularly useful when you want to optimize the memory usage of MCUboot to less than 16 kB by enabling cryptographic functionalities.

.. note::

>    The ``CONFIG_PM_PARTITION_SIZE_MCUBOOT`` option should be set in the child image's configuration to match the reduced size.

Author: In your suggestion, should the hanging ``_`` at the end be there?

>    The ``CONFIG_PM_PARTITION_SIZE_MCUBOOT`` option should be set in the child image's configuration to match the reduced size.

Author: Ah, okay - will revise accordingly

The :kconfig:option:`CONFIG_PM_PARTITION_SIZE_<CHILD_IMAGE>` Kconfig option allows you to specify the memory size for each child image's partition directly in the Kconfig file.
This allows for precise control over memory allocation which is crucial for system performance optimization.

Common variants of this Kconfig's option include the following:

> * ``CONFIG_PM_PARTITION_SIZE_MCUBOOT``: Sets the partition size for the MCUboot image.

Reviewer: About the policy: This should be a fixed link under links.txt

> * ``CONFIG_PM_PARTITION_SIZE_MCUBOOT``: Sets the partition size for the MCUboot image.

Reviewer: @kittydepa , yes, according to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/dev_model_and_contributions/documentation/styleguide.html#external-links

* :kconfig:option:`CONFIG_PM_PARTITION_SIZE_PROVISION`: Allocates space for the provision partition used in secure boot scenarios.
* :kconfig:option:`CONFIG_PM_PARTITION_SIZE_SETTINGS`: Defines the size of the settings storage partition, typically used for persistent configuration data.
* :kconfig:option:`CONFIG_PM_PARTITION_SIZE_SETTINGS_STORAGE`: Sets the size of the settings storage partition.
* :kconfig:option:`CONFIG_PM_PARTITION_SIZE_NVS_STORAGE`: Configures the size of the non-volatile storage partition.

Each variant must be adjusted according to the specific requirements and memory constraints of the project, ensuring optimal usage of available flash memory.

> For a full list of the :kconfig:option:`CONFIG_PM_PARTITION_SIZE`, see the related `Kconfig search results <https://docs.nordicsemi.com/bundle/ncs-latest/page/kconfig/index.html#!CONFIG_PM_PARTITION_SIZE>`_.

Reviewer 2: Is it on purpose that this specifically links to new docs? 
Also the link does not work for me. As you can see it does not show the search you try to link to.
I have tried to link to "unfinished" configurations before with no luck, so I would be interested to see if this is possible
![image](https://github.com/nrfconnect/sdk-nrf/assets/89006246/400be062-dc1a-4bd5-a193-e60a104bffda)

Author: I'll look into this - which browser were you using?

Reviewer 2: Firefox.
I can confirm that the link works on chrome

Reviewer: *cough* *cough* Also in case of this link, we need to add it to links.txt. *cough*

Author: Okay, will do

.. note::
  If the static configuration contains an entry for the ``app`` partition, this entry is ignored.

.. _ug_pm_config_dynamic:

Configuring dynamic partitions
==============================

For child images that use dynamic partition sizing, the Partition Manager allows you to use the ``CONFIG_PM_PARTITION_SIZE_<CHILD_IMAGE>`` Kconfig option to adjust the partition size.

> This is particularly useful when you want to optimize the memory usage of MCUboot to less than 16 kB by enabling cryptographic functionalities.

Reviewer 3: I would not mention particular size (16 kB) as it might be not achievable. Maybe 'the smallest possible size` or so.
The smallest possible size might vary form release to release.

.. note::

>    The ``CONFIG_PM_PARTITION_SIZE_<CHILD_IMAGE>`` option should be set in the child image's configuration to match the reduced size.

Reviewer 4: ```suggestion
   To match the reduced size, you need to set the ``CONFIG_PM_PARTITION_SIZE_<CHILD_IMAGE>`` Kconfig option in the child image's configuration.
```

Reviewer 4: +all CONFIG_PM_PARTITION_SIZE options should probably render as Kconfig options, though the option itself doesn't appear in the Kconfig search

   For example, ``CONFIG_PM_PARTITION_SIZE_MCUBOOT`` must be set for the MCUboot child image.

The ``CONFIG_PM_PARTITION_SIZE_<CHILD_IMAGE>`` Kconfig option allows you to specify the memory size for each child image's partition directly in the Kconfig file.
This allows for precise control over memory allocation which is crucial for system performance optimization.

Common variants of this Kconfig's option include the following:

> * ``CONFIG_PM_PARTITION_SIZE_MCUBOOT``: Sets the partition size for the MCUboot image, and only used for dynamic partition maps.

Reviewer 4: ```suggestion
* ``CONFIG_PM_PARTITION_SIZE_MCUBOOT`` - Sets the partition size for the MCUboot image, and only used for dynamic partition maps.
```

  You can read more about this option in the `MCUboot Kconfig option documentation`_.

> * :kconfig:option:`CONFIG_PM_PARTITION_SIZE_PROVISION`: Allocates space for the provision partition used in secure boot scenarios.
> 
> * :kconfig:option:`CONFIG_PM_PARTITION_SIZE_SETTINGS`: Defines the size of the settings storage partition, typically used for persistent configuration data.
> 
> * :kconfig:option:`CONFIG_PM_PARTITION_SIZE_SETTINGS_STORAGE`: Sets the size of the settings storage partition.
> 
> * :kconfig:option:`CONFIG_PM_PARTITION_SIZE_NVS_STORAGE`: Configures the size of the non-volatile storage partition.

Reviewer 4: ```suggestion
* :kconfig:option:`CONFIG_PM_PARTITION_SIZE_PROVISION` - Allocates space for the provision partition used in secure boot scenarios.

* :kconfig:option:`CONFIG_PM_PARTITION_SIZE_SETTINGS` - Defines the size of the settings storage partition, typically used for persistent configuration data.

* :kconfig:option:`CONFIG_PM_PARTITION_SIZE_SETTINGS_STORAGE` - Sets the size of the settings storage partition.

* :kconfig:option:`CONFIG_PM_PARTITION_SIZE_NVS_STORAGE` - Configures the size of the non-volatile storage partition.
```

> Each variant must be adjusted according to the specific requirements and memory constraints of the project, ensuring optimal usage of available flash memory.

Reviewer 4: ```suggestion
You must adjust each variant according to the specific requirements and memory constraints of the project, ensuring optimal usage of available flash memory.
```
